### PR TITLE
feat: Required Slot Responses

### PR DESCRIPTION
**Pull request by michaelmyers**
This furthers our support for slot filling by adding a `slotElicitationContentKey ` to the slot value which will be used when the slot is not provided by the user to retrieve a response from the content dictionary.

### CODE REVIEW

File: packages/stentor-utils/src/compilers/compileJSONPaths.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ function compileString(value: string, object: object, replaceWhenUndefined: bool
     // Set exit condition to be when the results are null
     while (result !== null) {
         // index 1 is the capture
-        const captured = result[1];
+        const captured = result[1].trim();
```
Comment (chrsdietz) [inline packages/stentor-utils/src/compilers/compileJSONPaths.ts:15]: Is there always guaranteed to be a value at `result[1]`?

File: packages/stentor-guards/src/isHandler.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { Handler, Intent } from "stentor-models";
  *
  * @public
  */
-export function isHandler(props: Handler | Intent): props is Handler {
+export function isHandler(props: Handler | Intent | object | number | boolean): props is Handler {
```
Comment (OpenDog) [inline packages/stentor-guards/src/isHandler.ts:9]: How can a number be a handler candidate? And what object if not handler or intent?

File: packages/stentor-utils/src/compilers/compileJSONPaths.ts (hunk 1 above)
Comment (michaelmyers) [reply packages/stentor-utils/src/compilers/compileJSONPaths.ts:15]: I am pretty sure, the regex has a capture group in it which will be at `result[1]` so if the regex matches at all (`result !== null`) then the capture group should have a value.

**Comment by OpenDog** [comment]
This is great! I will try to create a Pepsi version and use it for product/flavor/zip. Then we can see it in action.

File: packages/stentor-guards/src/isHandler.ts (hunk 2 above)
Comment (michaelmyers) [reply packages/stentor-guards/src/isHandler.ts:9]: Really never, I only added those to make it easier to in cases where a Handler is a potential input to a function and to appease TypeScript, I add those to simplify the logic needed.  See [here](https://github.com/stentorium/stentor/pull/518/files#diff-a5fd25c20dd0ac3db6349d85b9fe7cb4R30) where `isHandler(content)` and `content` can potentially be another `object` or an array (also typed as an `object`).  I don't have to add those extra checks before calling `isHandler`

**Pull request merged**